cli/src/semgrep/commands/ci.py

```diff
@@ -293,134 +267,125 @@ def ci(
+            scan_handler.start_scan(metadata_dict)
+
+            config = (scan_handler.rules,)
+
+            scan_env.add_row(
+                "server",
+                "-",
+                f"logged in as [bold]{scan_handler.deployment_name}[/bold] on [bold]{state.env.semgrep_url}[/bold]",
+            )
+
+    except Exception as e:
+        import traceback
+
+        traceback.print_exc()
+        logger.info(f"Could not start scan {e}")
+        sys.exit(FATAL_EXIT_CODE)
+
+    if requested_engine is None:
+        requested_engine = EngineType.get_default(
+            scan_handler=scan_handler, git_meta=metadata
+        )
+
+    if dataflow_traces is None:
+        dataflow_traces = requested_engine.has_dataflow_traces
 
     console.print(Title("Scan Environment", order=2))
     console.print(Padding(scan_env, (0, 2)))
 
+    if requested_engine.is_pro:
+        console.print(Padding(Title("Engine", order=2), (1, 0, 0, 0)))
+        if requested_engine.check_if_installed():
+            console.print(
+                f"  Using Semgrep Pro Version: [bold]{requested_engine.get_pro_version()}[/bold]"
+            )
+            console.print(
+                f"  Installed at [bold]{requested_engine.get_binary_path()}[/bold]"
+            )
+        else:
+            run_install_semgrep_pro()
+
     try:
-        with fix_head_if_github_action(metadata):
-            if scan_handler:
-                metadata_dict = metadata.to_dict()
-                metadata_dict["is_sca_scan"] = supply_chain
-
-            console.print(Title("Scan Status"))
-
-            try:
-                logger.info("Fetching configuration from semgrep.dev")
-                # Note this needs to happen within fix_head_if_github_action
-                # so that metadata of current commit is correct
-                if scan_handler:
-                    proj_config = ProjectConfig.load_all()
-                    metadata_dict = {**metadata_dict, **proj_config.to_dict()}
-                    scan_handler.fetch_and_init_scan_config(metadata_dict)
-                    scan_handler.start_scan(metadata_dict)
-
-                    logger.info(f"Authenticated as {scan_handler.deployment_name}")
-                    config = (scan_handler.rules,)
-            except Exception as e:
-                import traceback
-
-                traceback.print_exc()
-                logger.info(f"Could not start scan {e}")
-                sys.exit(FATAL_EXIT_CODE)
-
-            # Run Semgrep Pro when available
-            # By default, this is controlled by the toggle. When it is off, run
-            # the OSS engine. When it is on, for full scans run using interfile
-            # analysis and diff scans run using the pro languages
-            is_full_scan = metadata.merge_base_ref is None
-            engine = EngineType.OSS
-            if scan_handler and scan_handler.deepsemgrep:
-                engine = (
```

Comment:
Need to check later that this behavior is still consistent post change: `--pro` becomes pro-lang if not full scan and pro-interfile if diff scan